keps/sig-auth/541-external-credential-providers/README.md

```diff
@@ -493,6 +493,114 @@ credentials).
 PEM format. The certificate must be valid at the time of execution. These
 credentials are used for mTLS handshakes.
 
+### Metrics
+
+As discussed [below](#rollout-upgrade-and-rollback-planning), there are 4
+primary metrics used by this feature set.
+
+```golang
+var (
+  execPluginCertTTL = k8smetrics.NewGaugeFunc(
+    k8smetrics.GaugeOpts{
+      Name: "rest_client_exec_plugin_ttl_seconds",
+      Help: "Gauge of the shortest TTL (time-to-live) of the client " +
+        "certificate(s) managed by the auth exec plugin. The value " +
+        "is in seconds until certificate expiry (negative if " +
+        "already expired). If auth exec plugins are unused or manage no " +
+        "TLS certificates, the value will be +INF.",
+    },
+    func() float64 {
+      if execPluginCertTTLAdapter.e == nil {
+        return math.Inf(1)
+      }
+      return execPluginCertTTLAdapter.e.Sub(time.Now()).Seconds()
+    },
+  )
+
+  execPluginCertRotation = k8smetrics.NewHistogram(
+    &k8smetrics.HistogramOpts{
+      Name: "rest_client_exec_plugin_certificate_rotation_age",
+      Help: "Histogram of the number of seconds the last auth exec " +
+        "plugin client certificate lived before being rotated. " +
+        "If auth exec plugin client certificates are unused, " +
+        "histogram will contain no data.",
+      // There are three sets of ranges these buckets intend to capture:
+      //   - 10-60 minutes: captures a rotation cadence which is
+      //     happening too quickly.
+      //   - 4 hours - 1 month: captures an ideal rotation cadence.
+      //   - 3 months - 4 years: captures a rotation cadence which is
+      //     is probably too slow or much too slow.
+      Buckets: []float64{
+        600,     // 10 minutes
+        1800,    // 30 minutes
+        3600,    // 1  hour
+        14400,   // 4  hours
+        86400,   // 1  day
+        604800,  // 1  week
+        2592000,   // 1  month
+        7776000,   // 3  months
+        15552000,  // 6  months
+        31104000,  // 1  year
+        124416000, // 4  years
+      },
+    },
+  )
+
+  execPluginCalls = k8smetrics.NewCounterVec(
+    &k8smetrics.CounterOpts{
+      Name: "rest_client_exec_plugin_calls",
+      Help: "Number of calls to an exec plugin.",
+    },
+    []string{},
+  )
+
+  execPluginFailedCalls = k8smetrics.NewCounterVec(
```

Comment:
Is there anyway we could collapse these `rest_client_exec_plugin_calls` and `rest_client_exec_plugin_failed_calls` metrics into a single metric with an `"exitCode"` label to indicate whether this was a successful call to the exec plugin?

Reply:
@ehashman - do you have any guidance to give on if we could collapse `rest_client_exec_plugin_calls` and `rest_client_exec_plugin_failed_calls` into a single metric with an `"exitCode"` label? I feel like this would just clean things up a bit.

Reply:
Yeah that's feasible, although I think the standard label would be `code`. See e.g. https://github.com/kubernetes/apiserver/blob/4cca99e7fbc563a06b3c505af12bfb90b79d3bcf/pkg/endpoints/metrics/metrics.go#L76-L87 for some inspiration - @logicalhan will probably have opinions if he gets a chance to look

Reply:
Thanks @ehashman - really appreciate your feedback here. I gave this simplification a go in 191094e76fd16bbc986b351104a7a5ffd81b4740.